htdocs/core/lib/date.lib.php

```diff
@@ -179,11 +179,12 @@ function convertSecondToTime($iSecond, $format='all', $lengthOfDay=86400, $lengt
 	{
 		if ((int) $iSecond === 0) return '0';	// This is to avoid having 0 return a 12:00 AM for en_US
 
-        $sTime='';
-        $sDay=0;
-        $sWeek=0;
+		$sTime='';
+		$sDay=0;
+		$sWeek=0;
 
-		if ($iSecond >= $lengthOfDay)
+		// if $lengthOfDay != 86400 allhour and allhourmin return wrong value
```

Comment:
Old code looks good to me. Can you provide an example of input and output data and what you expect ?

Reply:
If we have 
convertSecondToTime(100000, 'allhour', 86400) return 27 h
convertSecondToTime(100000, 'allhour', 28800) return 75 h
This is the expected value. If you work in base 28800, it means you want to know the duration of a task of 100 000 seconds when you work 8h per day. 100 000 seconds is 27.77 days, so if day if 8h only instead of 24h, you need to wait 75h to have task finished (instead of 27h if you work 24/24).